### DISCUSSION TITLE
useFieldArray not working when append/insert called multiple times quickly

### DISCUSSION THREAD

**Discussion by harshzalavadiya** [react-hook-form/react-hook-form#5324] [Q&A] [answered]
**Describe the bug**
When using fieldArray in react-hook-form@7 if append is called multiple times it does not work as expected, I don't know weather this was designed intentionally since it works when timeout is added before calling

use case: In our application I have to create multiple dropdowns based on parsed excel sheet to map fields, so I can't provide value as defaultValues, I have to create N fields at runtime all at once

**To Reproduce**
see `useState` method of codesandbox

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-05z9z?file=/src/index.js

**Expected behavior**
It should create multiple fields

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. iOS] Linux
- Browser [e.g. chrome, safari] chrome and firefox
- Version [e.g. 22] chrome@90 and firefox@88

**Smartphone (please complete the following information):**

**Additional context**
If you try it with `insert()` from `useFieldArray` internally it creates fields but `fields` object from `useFieldArray` only shows 1 item as above

**Comment by bluebill1049** [comment]
<img width="992" alt="Screen Shot 2021-05-25 at 7 25 20 pm" src="https://user-images.githubusercontent.com/10513364/119473884-f3e7ea00-bd8e-11eb-98a6-905daaca2ed8.png">

if you need to append multiple rows, use array syntax.

**Comment by bluebill1049** [comment]
<img width="952" alt="Screen Shot 2021-05-25 at 7 26 31 pm" src="https://user-images.githubusercontent.com/10513364/119474051-1da11100-bd8f-11eb-9655-153a1149ce0b.png">

**Comment by harshzalavadiya** [comment, marked as answer]
@bluebill1049 thanks for quick response,

Is there any way to add multiple fields at once after form is initialized, for example if I want to append 5 fields at once into array, or adding timeout is only way?, since this fields are added after form is initialized and size is unknown

Answer: https://github.com/react-hook-form/react-hook-form/discussions/5324#discussioncomment-780219

**Reply by bluebill1049** [reply]
```
append([
  {},
  {},
  {},
])
```

?

**Reply by harshzalavadiya** [reply]
My bad it takes array as arguement as well.

Anyways react-hook-form is amazing library and thanks for all your work :)

**Reply by bluebill1049** [reply]
no worries. ❤️

**Comment by henrikq** [comment]
First of, thanks for a amazing library. 

That said, Is there a good reason for this limitation? It seems like a junior-footgun, and a potential major inconvenience for the more senior dev. The solution the docs proposes:

```
// ✅ The following is correct and second action is triggered after next render
handleChange={() => {
  append({ test: 'test' });
}}

React.useEffect(() => {
  if (fields.length === 2) {
    remove(0);
  }
}, [fields])
```

is not very comforting. It requires you to split logic that may make sense together into disparate places in code. As is, It also comes with a rendering artifact; that is, for a single frame you are going to see two fields render for the example, although this may be fixable by using `useLayoutEffect`?

I don't know if this suggestion has allready been rejected, but maybe it at-least should be possible to update using a [functional update](https://reactjs.org/docs/hooks-reference.html#functional-updates)?
